src/select/SelectMenu.ts

```diff
@@ -1,77 +1,54 @@
-import { createHook, createComponent } from "reakit-system";
+import { usePopover, PopoverHTMLProps } from "reakit";
+import { SelectStateReturn } from "./SelectState";
 import { SELECT_KEYS } from "./__keys";
-
-export type SelectMenuOptions = Pick<
-  SelectStateReturn,
-  "setTypehead" | "selected" | "hide" | "show" | "visible"
-> & {
-  onChange?: (value: any) => void;
-};
-
-const useSelectMenu = createHook<SelectMenuOptions, BoxHTMLProps>({
+import { createComponent, createHook } from "reakit-system";
+import {
+  useComposite,
+  CompositeProps,
+  CompositeHTMLProps,
+} from "reakit/Composite";
+import { usePortalShortcut } from "./__utils";
+import { useForkRef } from "reakit-utils";
+
+export type SelectMenuOptions = CompositeProps &
+  Pick<
+    SelectStateReturn,
+    "setSelected" | "visible" | "values" | "currentId" | "move"
+  > & {
+    modal?: boolean;
+  };
+
+export type SelectMenuHTMLProps = CompositeHTMLProps & PopoverHTMLProps;
+
+const useSelectMenu = createHook<SelectMenuOptions, SelectMenuHTMLProps>({
   name: "SelectMenu",
+  compose: [useComposite, usePopover],
   keys: SELECT_KEYS,
+  useOptions({ modal = false, ...options }) {
+    return { modal, ...options };
+  },
 
   useProps(
-    { setTypehead, selected, onChange, hide, show, visible },
-    { ...htmlProps },
+    { visible, move, values, currentId },
+    { ref: htmlRef, style: htmlStyle, ...htmlProps },
   ) {
-    const keyClear = React.useRef<any>(null);
-    const [typed, setTyped] = React.useState("");
-
-    React.useEffect(() => {
-      onChange && onChange(selected);
-    }, [selected]);
-
-    const onKeyDown = React.useMemo(() => {
-      return createOnKeyDown({
-        stopPropagation: true,
-        keyMap: () => ({
-          Escape: hide,
-          ArrowUp: show,
-          ArrowDown: show,
-        }),
-      });
-    }, []);
+    const ref = React.useRef<HTMLElement>(null);
 
-    const clearKeyStrokes = () => {
-      setTypehead(typed);
-
-      if (keyClear.current) {
-        clearTimeout(keyClear.current);
-        keyClear.current = null;
-      }
-
-      keyClear.current = setTimeout(() => {
-        setTyped("");
-        keyClear.current = null;
-      }, 800);
-    };
+    usePortalShortcut(ref, { values, currentId, move });
 
     React.useEffect(() => {
-      if (typed !== "") {
-        clearKeyStrokes();
+      if (values) {
+        const firstItem = values[0];
+        firstItem && move?.(firstItem.id);
       }
-    }, [typed]);
-
-    const handleOnKeyPress = (e: React.KeyboardEvent) => {
-      e.persist();
-      // skip the enter key
-      if (e.key === "Enter") return;
-      setTyped(prev => prev + e.key);
-    };
+    }, [visible]);
 
     return {
-      role: "button",
-      "aria-expanded": visible,
-      "aria-haspopup": "listbox",
-      onKeyDown,
-      onKeyPress: handleOnKeyPress,
+      ref: useForkRef(ref, htmlRef),
+      role: "listbox",
+      style: htmlStyle,
```

Comment:
you can remove this completely !!

Reply:
Which one?

Reply:
Oops, selected everything 🤦 
the `style`

Reply:
Does not seem to be working without htmlStyle

Reply:
You need to remove the destructuring above too.

Reply:
If you have doubt, can I push a commit here 🤔

Reply:
![image](https://user-images.githubusercontent.com/39694575/91559879-79e91880-e956-11ea-85fb-b28d19de5aaf.png)

Don't desctructure and just let it pass, that's what I said.

Reply:
Ohhhh okay.